packages/js-sdk/src/templates/statefulCodeInterpreter.ts

```diff
@@ -0,0 +1,268 @@
+   *
+   * await sandbox.keepAlive(300 * 1000)
+   * await sandbox.close()
+   *
+   * const reconnectedSandbox = await Sandbox.reconnect(sandboxID)
+   * ```
+   */
+  static async reconnect<S extends typeof CodeInterpreterV2>(this: S, sandboxID: string): Promise<InstanceType<S>>
+  /**
+   * Reconnects to an existing Sandbox.
+   * @param opts Sandbox options
+   * @returns Existing Sandbox
+   *
+   * @example
+   * ```ts
+   * const sandbox = await Sandbox.create()
+   * const sandboxID = sandbox.id
+   *
+   * await sandbox.keepAlive(300 * 1000)
+   * await sandbox.close()
+   *
+   * const reconnectedSandbox = await Sandbox.reconnect({
+   *   sandboxID,
+   * })
+   * ```
+   */
+  static async reconnect<S extends typeof CodeInterpreterV2>(this: S, opts: Omit<SandboxOpts, 'id' | 'template'> & { sandboxID: string }): Promise<InstanceType<S>>
+  static async reconnect<S extends typeof CodeInterpreterV2>(this: S, sandboxIDorOpts: string | Omit<SandboxOpts, 'id' | 'template'> & { sandboxID: string }): Promise<InstanceType<S>> {
+    let id: string
+    let opts: SandboxOpts
+    if (typeof sandboxIDorOpts === 'string') {
+      id = sandboxIDorOpts
+      opts = {}
+    } else {
+      id = sandboxIDorOpts.sandboxID
+      opts = sandboxIDorOpts
+    }
+
+    const sandboxIDAndClientID = id.split('-')
+    const sandboxID = sandboxIDAndClientID[0]
+    const clientID = sandboxIDAndClientID[1]
+    opts.__sandbox = { sandboxID, clientID, templateID: 'unknown' }
+
+    const sandbox = new this(opts) as InstanceType<S>
+    await sandbox._open({ timeout: opts?.timeout })
+
+    sandbox.jupyterKernelID = await sandbox.getKernelID()
+    return sandbox
+  }
+
+  private static sendExecuteRequest(code: string) {
+    const msg_id = randomBytes(16).toString('hex')
+    const session = randomBytes(16).toString('hex')
+    return {
+      header: {
+        msg_id: msg_id,
+        username: 'e2b',
+        session: session,
+        msg_type: 'execute_request',
+        version: '5.3',
+      },
+      parent_header: {},
+      metadata: {},
+      content: {
+        code: code,
+        silent: false,
+        store_history: false,
+        user_expressions: {},
+        allow_stdin: false,
+      },
+    }
+  }
+
+  async execPython(
+    code: string,
+    onStdout?: (out: ProcessMessage) => Promise<void> | void,
+    onStderr?: (out: ProcessMessage) => Promise<void> | void,
+  ) {
+    let resolve: () => void
+
+    const finished = new Promise<void>((r) => {
+      resolve = () => r()
+    })
+    const result: Result = {
+      stdout: [],
+      stderr: [],
+      display_data: [],
+    }
+
+    // @ts-ignore
+    const ws = await this._connectKernel(result, resolve, onStdout, onStderr)
+    ws.send(JSON.stringify(CodeInterpreterV2.sendExecuteRequest(code)))
+    await finished
+
+    ws.close()
+    return result
+  }
+
+  private async getKernelID() {
+    return await this.filesystem.read('/root/.jupyter/kernel_id')
```

Comment:
```suggestion
    return (await this.filesystem.read('/root/.jupyter/kernel_id')).trim();
```

There was an extra newline at the end on cloudflare workers durable objects.

Reply:
Thanks @lawrencecchen , all the fixes will be incorporated in this repo https://github.com/e2b-dev/code-interpreter